source/d3d9/runtime_d3d9.cpp

```diff
@@ -1394,4 +1767,164 @@ namespace reshade::d3d9
+			{
+				desc.Format = format;
+				break;
+			}
+		}
+
+		if (desc.Format == D3DFMT_UNKNOWN)
+		{
+			LOG(ERROR) << "Your graphics card is missing support for at least one of the 'INTZ', 'DF24' or 'DF16' texture formats. Cannot create depth replacement texture.";
+
+			return nullptr;
+		}
+
+		UINT width = desc.Width;
+		UINT height = desc.Height;
+
+		if (_disable_depth_buffer_size_restriction)
+		{
+			width = _width;
+			height = _height;
+		}
+
+		const HRESULT hr = _device->CreateTexture(width, height, 1, D3DUSAGE_DEPTHSTENCIL, desc.Format, D3DPOOL_DEFAULT, &depthstencil_replacement, nullptr);
+
+		if (!SUCCEEDED(hr))
+		{
+			LOG(ERROR) << "Failed to create depth replacement texture! HRESULT is '" << std::hex << hr << std::dec << "'.";
+
+			return nullptr;
+		}
+
+		return depthstencil_replacement;
+	}
+
+	// retrieve the depth buffer surface replacement ref, depending on the detection mode
+	com_ptr<IDirect3DSurface9> runtime_d3d9::get_depthstencil_replacement()
+	{
+		const auto it = _depth_clearing_table.find(_clear_idx);
+
+		if (it != _depth_clearing_table.end())
+		{
+			return it->second.depthstencil;
+		}
+
+		return _depthstencil_replacement;
+	}
+
+	// retrieve the depth buffer texture replacement ref, depending on the detection mode
+	com_ptr<IDirect3DTexture9> runtime_d3d9::get_depthstencil_texture()
+	{
+		const auto it = _depth_clearing_table.find(_clear_idx);
+
+		if (it != _depth_clearing_table.end())
+		{
+			return it->second.depthtexture;
+		}
+		return _depthstencil_texture;
+	}
+
+	// retrieve the best depth buffer clearing instance, depending on the options selected
+	int runtime_d3d9::get_best_preserve_starting_index(bool multi)
+	{
+		int result = multi ? -1 : 0;
+
+		for (const auto &it : _depth_clearing_table)
+		{
+			bool best = (it.second.vertices_count >= _db_vertices);
+			if (_auto_preserve_on_drawcalls)
+				best = (it.second.drawcall_count >= _db_drawcalls);
+
+			if (best)
+			{
+				_db_vertices = it.second.vertices_count;
+				_db_drawcalls = it.second.drawcall_count;
+			}
+
+			// only when multiple depth buffer replacement textures
+			if (best && multi)
+			{
+				if (_auto_preserve_on_vertices)
+				{
+					// compare with the max number of vertices from the previous frame. If too much difference, exclude this candidate
+					if (it.second.vertices_count < (_previous_best_vertices / 10))
+						best = false;
+				}
+				if (_auto_preserve_on_drawcalls)
+				{
+					// compare with the max number of drawcalls from the previous frame. If too much difference, exclude this candidate
+					if (it.second.drawcall_count < (_previous_best_drawcalls / 10))
+						best = false;
+				}
+			}
+
+			if (best)
+				result = it.first;
+		}
+
+		return result;
+	}
+
+	bool runtime_d3d9::check_depthstencil_size(com_ptr<IDirect3DSurface9> depthstencil)
```

Comment:
Everywhere this is called, `GetDesc` is called either before or soon after. So to avoid duplicated calls, just change this function signature to `bool check_depthstencil_size(const D3DSURFACE_DESC &desc);` and pass in the description in the call.